### PR TITLE
Clean up jakarta jaxb API and impl bundles

### PR DESCRIPTION
**Pull request by jhanders34**
- Update to use JAXB API and impl 3.0.1
- Remove module-info.class from the bundles
- Stop doing override now that the fix is in the 3.0.1 API jars as fixed with https://github.com/eclipse-ee4j/jaxb-api/issues/172
- Update the API and impl jar to make them multi release jars since they include multi release artifacts.

This completes:
    https://github.com/OpenLiberty/open-liberty/pull/12930/commits/56139ccfc17c976138868e29aa0533081273bf07#r451592082

### CODE REVIEW

**Comment by jhanders34** [comment]
#build

File: dev/io.openliberty.jakarta.xmlBinding.3.0/bnd.bnd [hunk 1]
```diff
@@ -31,16 +31,17 @@ DynamicImport-Package: \
 #Bundle-Activator: org.apache.geronimo.osgi.locator.Activator
 
 Include-Resource:\
-  @${repo;jakarta.xml.bind:jakarta.xml.bind-api;3.0.0;EXACT}!/!META-INF/maven/*, \
-  jakarta/xml/bind=${bin}/jakarta/xml/bind
+  @${repo;jakarta.xml.bind:jakarta.xml.bind-api;3.0.1;EXACT}!/!(META-INF/maven/*|module-info.class)
 # @${repo;org.apache.geronimo.specs:geronimo-jaxb_2.2_spec;1.0.1}!/org/apache/geronimo/osgi/locator/*, \
```
Comment (tevans78) [inline dev/io.openliberty.jakarta.xmlBinding.3.0/bnd.bnd]: Do you know what these commented out lines are for?

**Review by tevans78 — APPROVED** [review]
Only question was about some commented out lines which are not actually part of this PR

File: dev/io.openliberty.jakarta.xmlBinding.3.0/bnd.bnd (hunk 1 above)
Comment (jhanders34) [reply dev/io.openliberty.jakarta.xmlBinding.3.0/bnd.bnd]: The bnd originally was copied from the 2.2 version and these were determined not to be needed I assume.  @neuwerk would know for sure.

Comment (neuwerk) [reply dev/io.openliberty.jakarta.xmlBinding.3.0/bnd.bnd]: Yeah, they were just stale code from 2.2, I should have deleted them at the time.

**Review by neuwerk — APPROVED** [review]
Changes look good to me!

**Comment by jhanders34** [comment]
#libby

File: dev/io.openliberty.jakarta.xmlBinding.3.0/bnd.bnd (hunk 1 above)
Comment (jhanders34) [reply dev/io.openliberty.jakarta.xmlBinding.3.0/bnd.bnd]: I removed them.

**Pull request merged**